lib/vdc/BOVCollection.cpp

```diff
@@ -0,0 +1,529 @@
+const std::string BOVCollection::_yDim = "y";
+const std::string BOVCollection::_zDim = "z";
+const std::string BOVCollection::_timeDim = "t";
+
+const std::string BOVCollection::_byteFormatString = "BYTE";
+const std::string BOVCollection::_shortFormatString = "SHORT";
+const std::string BOVCollection::_intFormatString = "INT";
+const std::string BOVCollection::_floatFormatString = "FLOAT";
+const std::string BOVCollection::_doubleFormatString = "DOUBLE";
+
+BOVCollection::BOVCollection()
+: _time(_defaultTime), _dataFile(_defaultFile), _dataFormat(_defaultFormat), _variable(_defaultVar), _dataEndian(_defaultEndian), _centering(_defaultCentering), _byteOffset(_defaultByteOffset),
+  _divideBrick(_defaultDivBrick), _dataComponents(_defaultComponents), _tmpDataFormat(_defaultFormat), _tmpDataEndian(_defaultEndian), _tmpByteOffset(_defaultByteOffset), _gridSizeAssigned(false),
+  _formatAssigned(false), _brickOriginAssigned(false), _brickSizeAssigned(false), _dataEndianAssigned(false), _byteOffsetAssigned(false), _timeDimension(_timeDim)
+{
+    _dataFiles.clear();
+    _times.clear();
+    _gridSize = _defaultGridSize;
+    _tmpGridSize = _defaultGridSize;
+    _brickOrigin = _defaultOrigin;
+    _tmpBrickOrigin = _defaultOrigin;
+    _brickSize = _defaultBrickSize;
+    _tmpBrickSize = _defaultBrickSize;
+    _dataBricklets = _defaultBricklets;
+    _spatialDimensions = {_xDim, _yDim, _zDim};
+}
+
+int BOVCollection::Initialize(const std::vector<std::string> &paths)
+{
+    VAssert(paths.size() > 0);
+
+    int           rc;
+    std::ifstream header;
+    for (int i = 0; i < paths.size(); i++) {
+        std::cout << paths[i] << std::endl;
+        header.open(paths[i]);
+        if (header.is_open()) {
+            rc = _parseHeader(header);
+            if (rc < 0) {
+                SetErrMsg(("Error parsing BOV file " + paths[0]).c_str());
+                return -1;
+            }
+
+            rc = _validateParsedValues();
+            if (rc < 0) {
+                SetErrMsg("Inconsistency found in BOV token.");
+                return -1;
+            }
+
+            // Ensure we have the required tokens in the header
+            //
+            if (_dataFile == _defaultFile) { return _missingValueError(_dataFileToken); }
+            if (_dataFormat == _defaultFormat) { return _missingValueError(_formatToken); }
+            if (_gridSize == _defaultGridSize) { return _missingValueError(_gridSizeToken); }
+
+            std::cout << _dataFile << std::endl;
+            std::ifstream infile(_dataFile);
+            if (infile.bad()) {
+                SetErrMsg(("Failed to open BOV file " + _dataFile).c_str());
+                return -1;
+            }
+
+            _populateDataFileMap();
+        } else {
+            SetErrMsg(("Failed to open BOV file " + paths[0]).c_str());
+            return -1;
+        }
+        header.close();
+    }
+
+    return 0;
+}
+
+int BOVCollection::_parseHeader(std::ifstream &header)
+{
```

Comment:
Is this parser supposed to correctly recognize scientific notations? If no, please state it in the documentation. If yes, does it?

Reply:
Yes, it does.